README.md

```diff
@@ -41,33 +42,81 @@ To import in your React code:
 The most basic example of a component using this hook would be:
 
 ```
-import React from 'react'
-import SpeechRecognition, { useSpeechRecognition } from 'react-speech-recognition'
+import React from 'react';
+import SpeechRecognition, { useSpeechRecognition } from 'react-speech-recognition';
 
 const Dictaphone = () => {
-  const { transcript, resetTranscript } = useSpeechRecognition()
+  const { transcript, listening, resetTranscript } = useSpeechRecognition();
 
   if (!SpeechRecognition.browserSupportsSpeechRecognition()) {
-    return null
+    return <span>Browser doesn't support speech recognition.</span>;
   }
 
   return (
     <div>
+      <p>Microphone: {listening ? 'on' : 'off'}</p>
       <button onClick={SpeechRecognition.startListening}>Start</button>
       <button onClick={SpeechRecognition.stopListening}>Stop</button>
       <button onClick={resetTranscript}>Reset</button>
       <p>{transcript}</p>
     </div>
-  )
-}
-export default Dictaphone
+  );
+};
+export default Dictaphone;
 ```
 
 You can see more examples in the example React app attached to this repo. See [Developing](#developing).
 
+## Cross-browser example
+
+By default, speech recognition is not supported in all browsers, with the best native experience being available on desktop Chrome. To build a web app that supports voice input on _all_ browsers, it's recommended that you combine `react-speech-recognition` with a [speech recognition polyfill](docs/POLYFILLS.md). It currently supports polyfills for the following cloud providers:
+
+<div>
+  <a href="https://www.speechly.com/?utm_source=github">
+    <img src="docs/logos/speechly.png" width="200" alt="Speechly">
+  </a>
+  <img width="50"></img>
+  <a href="https://azure.microsoft.com/en-gb/services/cognitive-services/speech-to-text/">
+    <img src="docs/logos/microsoft.png" width="175" alt="Microsoft Azure Cognitive Services">
+  </a>
+</div>
+
+You can find the full guide for setting up a polyfill [here](docs/POLYFILLS.md). Alternatively, here is a quick (and free) example using Speechly:
+* Install `@speechly/speech-recognition-polyfill` in your web app
+* You will need a Speechly app ID. To get one of these, sign up for free with Speechly and follow [the guide here](https://docs.speechly.com/quick-start/). Note that you can skip the steps for creating a Speechly configuration
+* Here's a component for a push-to-talk button. The basic example above would also work fine.
+```
+import React, { useState } from 'react';
+import { createSpeechlySpeechRecognition } from '@speechly/speech-recognition-polyfill';
+import SpeechRecognition, { useSpeechRecognition } from 'react-speech-recognition';
+
+const appId = '<INSERT_SPEECHLY_APP_ID_HERE>';
```

Comment:
I'm debating whether to cut out some of the repetition here by writing the push-to-talk component just once in the basic example. Then the Speechly sections could just say something like "to make the basic example work cross-browser, add the following code before its component definition:"
```
import { createSpeechlySpeechRecognition } from '@speechly/speech-recognition-polyfill';

const appId = '<INSERT_SPEECHLY_APP_ID_HERE>';
const SpeechlySpeechRecognition = createSpeechlySpeechRecognition(appId);
SpeechRecognition.applyPolyfill(SpeechlySpeechRecognition);
```

On one hand, I like the push-to-talk example as it encourages good practice. On the other, I'm wary of having that as the first example anyone sees as it does contain a load of touch/mouse interaction handlers. I'll sleep on it to see which way feels more inviting for someone reading the README for the first time.